src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkGenRepoInfoFile.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.appengine.cloudsdk;
+
+import com.google.cloud.tools.appengine.api.AppEngineException;
+import com.google.cloud.tools.appengine.api.deploy.GenRepoInfoFile;
+import com.google.cloud.tools.appengine.api.deploy.GenRepoInfoFileConfiguration;
+import com.google.cloud.tools.appengine.cloudsdk.internal.args.GcloudArgs;
+import com.google.cloud.tools.appengine.cloudsdk.internal.process.ProcessRunnerException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Cloud SDK based implementation of {@link GenRepoInfoFile}.
+ */
+public class CloudSdkGenRepoInfoFile implements GenRepoInfoFile {
+
+  private CloudSdk sdk;
+
+  public CloudSdkGenRepoInfoFile(CloudSdk sdk) {
+    this.sdk = sdk;
+  }
+
+  @Override
+  public void generate(GenRepoInfoFileConfiguration configuration) {
+    List<String> arguments = new ArrayList<>();
+
+    arguments.add("gen-repo-info-file");
+    arguments.addAll(GcloudArgs.get("output-directory", configuration.getOutputDirectory()));
+    arguments.addAll(GcloudArgs.get("output-file", configuration.getOutputFile()));
+    arguments.addAll(GcloudArgs.get("source-directory", configuration.getSourceDirectory()));
+
+    try {
+      sdk.runAppCommand(arguments);
+    } catch (ProcessRunnerException pre) {
```

Comment:
does ProcessrunnerException also get thrown if the command terminates with an error code?